oi/CodeGen.cpp

```diff
@@ -565,28 +565,74 @@ void addStandardTypeHandlers(std::string& code) {
   )";
 }
 
-// TODO support thrift isset
-void getClassTypeHandler(const Class& c, std::string& code) {
+}  // namespace
+
+void CodeGen::getClassTypeHandler(const Class& c, std::string& code) {
   std::string funcName = "getSizeType";
+  std::string extras;
+
+  const Member* thriftIssetMember = nullptr;
+  if (const auto it = thriftIssetMembers_.find(&c);
+      it != thriftIssetMembers_.end()) {
+    thriftIssetMember = it->second;
+
+    extras += "\n  using thrift_data = apache::thrift::TStructDataStorage<" +
+              c.fqName() + ">;";
+
+    extras += (boost::format(R"(
+  static int getThriftIsset(const %1%& t, size_t i) {
+    if (&thrift_data::isset_indexes == nullptr) return -1;
+
+    auto idx = thrift_data::isset_indexes[i];
+    if (idx == -1) return -1;
+
+    return t.%2%.get(idx);
+  }
+)") % c.name() %
+               thriftIssetMember->name)
+                  .str();
+  }
+
+  size_t nonPaddingSize = 0;
```

Comment:
This name is a little misleading, as it does count padding elements in the middle of a struct - it's only excluding those at the end.

I think I had a suggestion for a new variable name, but then I asked metamate for more ideas and now my mind is just full of the absolute rubbish it came up with... I'm not going to post them so I don't pollute your brain too 😛